pipeline_steps/common.groovy

```diff
@@ -1587,6 +1587,46 @@ List getComponentChange(String baseBranch, String prBranch){
 void registerComponent(String component_text, String jiraProjectKey){
   def component = readYaml text: component_text
   createComponentGateTrigger(component["name"], component["repo_url"], jiraProjectKey)
+  createComponentSkeleton(component["name"], component["repo_url"], jiraProjectKey)
+}
+
+void createComponentSkeleton(String name, String repoUrl, String jiraProjectKey){
+  dir("${WORKSPACE}/${name}"){
+    clone_with_pr_refs(".", repoUrl.replace("https://github.com/", "git@github.com:"), "origin/master")
+    withEnv(
+      [
+        "ISSUE_SUMMARY=Add component skeleton to ${name}",
+        "ISSUE_DESCRIPTION=This issue was generated automatically as part of registering a new component.",
+        "LABELS=component-skeleton",
+        "JIRA_PROJECT_KEY=${jiraProjectKey}",
+        "TARGET_BRANCH=master",
```

Comment:
A component could very well not use master as a branch name. Is this a platform requirement we need to impose, something we need to adjust, or something we deal w/ later when we hit a component that isn't using master?

Reply:
I was going with assume it is and deal with problems later. I think it's a pretty strong convention, maybe it would make sense to be explicit about that.